components/nexus-repository/src/main/java/org/sonatype/nexus/repository/proxy/ProxyFacetSupport.java

```diff
@@ -171,7 +185,15 @@ public Content get(final Context context) throws IOException {
   @Override
   public void invalidateProxyCaches() {
     log.info("Invalidating proxy caches of {}", getRepository().getName());
-    cacheController.invalidateCache();
+    config.cacheToken = cacheController.invalidateCache();
+    final ConfigurationFacet configurationFacet = facet(ConfigurationFacet.class);
+    configurationFacet.writeSection(getRepository().getConfiguration(), CONFIG_KEY, config);
+    try {
+      configurationFacet.save();
+    }
+    catch (Exception e) {
+      throw Throwables.propagate(e);
+    }
```

Comment:
why?  this seems overly complex, and very NX2 like where its constantly saving configuration, that was not the intent of the repository configuration to be used like this.